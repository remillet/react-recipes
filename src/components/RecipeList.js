import React from 'react';
import PropTypes from 'prop-types';
import RecipeListItem from './RecipeListItem';

const RecipeList = (props) => {
  const { recipes, style, favorites } = props;

  return (
    <div style={style}>
      <ul className="list-reset">
        {' '}
        {recipes.map(recipe => (
          <RecipeListItem
            key={recipe.id}
            recipe={recipe}
            favorites={favorites}
            {...props}
          />
        ))}
      </ul>
    </div>
  );
};

RecipeList.defaultProps = {
  style: null,
  recipes: [],
  favorites: [],
};

RecipeList.propTypes = {
  recipes: PropTypes.arrayOf(Object),
  favorites: PropTypes.arrayOf(String),
  style: PropTypes.object,
};

export default RecipeList;
