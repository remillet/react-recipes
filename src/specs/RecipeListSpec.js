import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../components/RecipeList';

const testRecipeList = [
  {
    id: 1,
    name: 'Recipe list item one',
    category: 'Dessert',
  },
  {
    id: 2,
    name: 'Recipe list item two',
    category: 'Pasta',
  },
  {
    id: 3,
    name: 'Recipe list item three',
    category: 'Bread',
  },
];

const testFavorites = [
  1,
  3,
];

describe('<RecipeList />', () => {
  test('Should not break if no recipe list is passed/provided.', () => {
    const component = renderer.create(<RecipeList />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly', () => {
    const component = renderer.create(<RecipeList recipes={testRecipeList} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('The render favorited states correctly', () => {
    const component = renderer.create(
      <RecipeList recipes={testRecipeList} favorites={testFavorites} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
