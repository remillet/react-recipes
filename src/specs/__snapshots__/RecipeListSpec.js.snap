// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> Should not break if no recipe list is passed/provided. 1`] = `
<div
  style={null}
>
  <ul
    className="list-reset"
  >
     
  </ul>
</div>
`;

exports[`<RecipeList /> Should render correctly 1`] = `
<div
  style={null}
>
  <ul
    className="list-reset"
  >
     
    <li
      className="py1 border-bottom border-bottom-dashed pointer"
      onClick={[Function]}
    >
      <span
        aria-label="favorite"
        className="mr1"
        onClick={[Function]}
        role="img"
      >
        ⌑
      </span>
      <span>
        Recipe list item one
      </span>
      <span>
         - 
      </span>
      <span>
        Dessert
      </span>
    </li>
    <li
      className="py1 border-bottom border-bottom-dashed pointer"
      onClick={[Function]}
    >
      <span
        aria-label="favorite"
        className="mr1"
        onClick={[Function]}
        role="img"
      >
        ⌑
      </span>
      <span>
        Recipe list item two
      </span>
      <span>
         - 
      </span>
      <span>
        Pasta
      </span>
    </li>
    <li
      className="py1 border-bottom border-bottom-dashed pointer"
      onClick={[Function]}
    >
      <span
        aria-label="favorite"
        className="mr1"
        onClick={[Function]}
        role="img"
      >
        ⌑
      </span>
      <span>
        Recipe list item three
      </span>
      <span>
         - 
      </span>
      <span>
        Bread
      </span>
    </li>
  </ul>
</div>
`;

exports[`<RecipeList /> The render favorited states correctly 1`] = `
<div
  style={null}
>
  <ul
    className="list-reset"
  >
     
    <li
      className="py1 border-bottom border-bottom-dashed pointer"
      onClick={[Function]}
    >
      <span
        aria-label="favorite"
        className="mr1"
        onClick={[Function]}
        role="img"
      >
        ⭐
      </span>
      <span>
        Recipe list item one
      </span>
      <span>
         - 
      </span>
      <span>
        Dessert
      </span>
    </li>
    <li
      className="py1 border-bottom border-bottom-dashed pointer"
      onClick={[Function]}
    >
      <span
        aria-label="favorite"
        className="mr1"
        onClick={[Function]}
        role="img"
      >
        ⌑
      </span>
      <span>
        Recipe list item two
      </span>
      <span>
         - 
      </span>
      <span>
        Pasta
      </span>
    </li>
    <li
      className="py1 border-bottom border-bottom-dashed pointer"
      onClick={[Function]}
    >
      <span
        aria-label="favorite"
        className="mr1"
        onClick={[Function]}
        role="img"
      >
        ⭐
      </span>
      <span>
        Recipe list item three
      </span>
      <span>
         - 
      </span>
      <span>
        Bread
      </span>
    </li>
  </ul>
</div>
`;
